disasm-test/cpp/smallprog.c

```diff
@@ -0,0 +1,42 @@
+/* #include <stdint.h> */
```

Comment:
Silly question: what actually tests this code? I was under the impression that `disasm-test` was ingesting `.ll` files, not `.c` files, so the lack of a corresponding `smallprog.ll` file surprises me.

Reply:
I hadn't generate an `.ll` file: I assumed that it might be different for different clang versions.

Currently this isn't part of any tests; I probably should have marked this PR as draft, my main goal was to drive awareness of this regression but I haven't had time to dive into it further.  I need to take care of a couple of other tasks first, and hopefully I will be able to extend this soon, unless anyone else happens to get there first.